starter-code/src/clue.js

```diff
@@ -1,94 +1,205 @@
+        occupation:   "Entrepreneur"
+        },
+        {first_name:   "Doctor",
+        last_name:    "Orchid",
+        color:        "white",
+        description:  "PhD in plant toxicology. Adopted daughter of Mr. Boddy",
+        age:          26,
+        image:        "http://www.radiotimes.com/uploads/images/Original/111967.jpg",
+        ocupation:   "Scientist"},
+        {
+        first_name:   "Victor",
+        last_name:    "Plum",
+        color:        "purple",
+        description:  "Billionare video game designer",
+        age:          22,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg",
+        occupation:   "Designer"
+        },
+
+        {first_name:   "Kasandra",
+        last_name:    "Scarlet",
+        color:        "red",
+        description:  "She is an A-list movie star with a dark past",
+        age:          31,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+        occupation:   "Actor"
+        },
+
+        {
+        first_name:   "Eleanor",
+        last_name:    "Peacock",
+        color:        "blue",
+        description:  "She is from a wealthy family and uses her status and money to earn popularity",
+        age:          36,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        occupation:   "Socialité"   
+        },
+
+        {
+        first_name:   "Jack",
+        last_name:    "Mustard",
+        color:        "yellow",
+        description:  "He is a former football player who tries to get by on his former glory",
+        age:          62,
+        image:        "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+        occupation:   "Retired Football player",  
+        }
+];
+
 
 // Rooms' Collection
-var roomsArray = [];
+var roomsArray = ["Dinning Room",
+    "Conservatory",
+    "Kitchen",
+    "Study",
+    "Library",
+    "Billiard Room",
+    "Lounge",
+    "Ballroom",
+    "Hall",
+    "Spa",
+    "Living Room",
+    "Observatory",
+    "Theater",
+    "Guest House",
+    "Patio"];
 
 // Weapons Collection
-var weaponsArray = [];
+var weaponsArray = [
+    {name: "rope"        ,   weight: 10},
+    {name: "knife"       ,   weight: 8},
+    {name: "candlestick" ,   weight: 2},
+    {name: "dumbbell"  ,   weight: 30},
+    {name: "poison"   ,   weight: 2},
+    {name: "axe"    ,   weight: 15},
+    {name: "bat"     ,   weight: 13},
+    {name: "trophy"      ,   weight: 25},
+    {name: "pistol"      ,   weight: 20}
+];
+
+function randomSelector (arrayEntry) {
+    return arrayEntry[Math.floor(Math.random() * arrayEntry.length)];
+};
+
+
+// var misteryEnvelope;
+
+function pickMistery(){
+    
+    misteryEnvelope = [ randomSelector(charactersArray)  , randomSelector(weaponsArray), randomSelector(roomsArray) ];
+    return  misteryEnvelope;
+    
+};
+
+var misteryEnvelope = pickMistery();
```

Comment:
To explain why you need to write 'var' again here : 
the var misteryEnvelope = pickMistery(); isn't the same one as the one you used earlier inside the pickMistery function : this one has a global scope and is used to store the array returned by the function. The one inside the function has a function scope, you could have called it myMistery or any other name and it would have worked perfectly well :)